packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -89,16 +96,66 @@ class BottomSheet extends Component {
 		Dimensions.addEventListener( 'change', this.onDimensionsChange );
 	}
 
-	keyboardWillShow( e ) {
+	keyboardShow( e ) {
+		if ( ! this.props.isVisible ) {
+			return;
+		}
+
 		const { height } = e.endCoordinates;
+		this.keyboardHeight = height;
+		this.performKeyboardLayoutAnimation( e );
+		this.onSetMaxHeight();
+		this.props.onKeyboardShow?.();
+	}
 
-		this.setState( { keyboardHeight: height }, () =>
-			this.onSetMaxHeight()
-		);
+	keyboardHide( e ) {
+		if ( ! this.props.isVisible ) {
+			return;
+		}
+
+		this.keyboardHeight = 0;
+		this.performKeyboardLayoutAnimation( e );
+		this.onSetMaxHeight();
+		this.props.onKeyboardHide?.();
+	}
+
+	performKeyboardLayoutAnimation( event ) {
+		const { duration, easing } = event;
+
+		if ( duration && easing ) {
+			const animationConfig = {
+				// We have to pass the duration equal to minimal accepted duration defined here: RCTLayoutAnimation.m
+				duration: duration > 10 ? duration : 10,
+				type: LayoutAnimation.Types[ easing ] || 'keyboard',
+			};
+			const layoutAnimation = {
+				duration: animationConfig.duration,
+				update: animationConfig,
+				create: {
+					...animationConfig,
+					property: LayoutAnimation.Properties.opacity,
+				},
+				delete: {
+					...animationConfig,
+					property: LayoutAnimation.Properties.opacity,
+				},
+			};
+			LayoutAnimation.configureNext( layoutAnimation );
+			this.lastLayoutAnimation = layoutAnimation;
+		} else {
```

Comment:
I thought the original note in your other branch that this code came form React Native's internal `KeyboardAvoidingView` was helpful, personally. It may be worth adding the comment that is the origin of this code.

Reply:
I decided to remove it because I wasn't really applying the same code but I agree that it would be helpful to keep the reference 👍 , I reverted the change in this [commit](https://github.com/WordPress/gutenberg/pull/30291/commits/7b865ade0da951d19a977c925a9655a112b0a6dd).